src/main/java/com/adobe/epubcheck/xml/XMLParser.java

```diff
@@ -135,13 +136,33 @@ public void process()
       }
 
       // Check encoding
-      String encoding = EncodingSniffer.sniffEncoding(buffered);
-      if (encoding != null && !encoding.equals("UTF-8") && !encoding.equals("UTF-16"))
+      // If the result is null, the XML parser will must parse it as UTF-8
+      String encoding = XMLEncodingSniffer.sniffEncoding(buffered);
+      if (encoding != null && !encoding.equals("UTF-8"))
       {
-        report.message(MessageId.CSS_003, EPUBLocation.of(context), encoding);
+        if (encoding.equals("UTF-16"))
+        {
+          // XHTML requires UTF-8, UTF-16 is reported as an error
```

Comment:
Is this correct? We don't require it in the epub spec and I can't find it in the HTML.

Reply:
It's specified in the [Encoding Standard](https://encoding.spec.whatwg.org/#names-and-labels), normatively referenced by HTML:

> Authors must use the [UTF-8](https://encoding.spec.whatwg.org/#utf-8) [encoding](https://encoding.spec.whatwg.org/#encoding) and must use the [ASCII case-insensitive](https://infra.spec.whatwg.org/#ascii-case-insensitive) "utf-8" [label](https://encoding.spec.whatwg.org/#label) to identify it.

And is made explicit in HTML section "[Specifying the document's character encoding](https://html.spec.whatwg.org/multipage/semantics.html#charset)":

> The Encoding standard requires use of the [UTF-8](https://encoding.spec.whatwg.org/#utf-8) [character encoding](https://encoding.spec.whatwg.org/#encoding) and requires use of the "utf-8" [encoding label](https://encoding.spec.whatwg.org/#label) to identify it. (…) Regardless of whether a [character encoding declaration](https://html.spec.whatwg.org/multipage/semantics.html#character-encoding-declaration) is present or not, the actual [character encoding](https://dom.spec.whatwg.org/#concept-document-encoding) used to encode the document must be [UTF-8](https://encoding.spec.whatwg.org/#utf-8). [[ENCODING]](https://html.spec.whatwg.org/multipage/references.html#refsENCODING)

Reply:
I'll resolve the conversation and merge the PR… (we can always open a new issue if I missed something).